juriscraper/opinions/united_states/state/ill.py

```diff
@@ -1,61 +1,80 @@
-# Contact: webmaster@illinoiscourts.gov, 217-558-4490, 312-793-3250
-# History:
-#   2013-08-16: Created by Krist Jin
-#   2014-12-02: Updated by Mike Lissner to remove the summaries code.
-#   2016-02-26: Updated by arderyp: simplified thanks to new id attribute identifying decisions table
-#   2016-03-27: Updated by arderyp: fixed to handled non-standard formatting
+# -*- coding: utf-8 -*-
+"""
+Contact: webmaster@illinoiscourts.gov, 217-558-4490, 312-793-3250
+ History:
+   2013-08-16: Created by Krist Jin
+   2014-12-02: Updated by Mike Lissner to remove the summaries code.
+   2016-02-26: Updated by arderyp: simplified thanks to new id attribute identifying decisions table
+   2016-03-27: Updated by arderyp: fixed to handled non-standard formatting
+   2021-11-02: Status validation based on Amendment to Rule 23 rulings
+               https://chicagocouncil.org/illinois-supreme-court-amendment-to-rule-23/
+"""
 
-from juriscraper.OpinionSite import OpinionSite
-from juriscraper.lib.string_utils import convert_date_string
+from juriscraper.OpinionSiteLinear import OpinionSiteLinear
+from juriscraper.AbstractSite import logger
+from juriscraper.lib.html_utils import (
+    get_row_column_links,
+    get_row_column_text,
+)
+import datetime
+import re
 
 
-class Site(OpinionSite):
+class Site(OpinionSiteLinear):
     def __init__(self, *args, **kwargs):
         super(Site, self).__init__(*args, **kwargs)
         self.court_id = self.__module__
-        self.url = "http://www.illinoiscourts.gov/Opinions/recent_supreme.asp"
-        # Sometimes court forgets pdf link in 5th cell.
-        # So, only process rows with a link in 5th cell.
-        limiter = 'contains(./td[5]//a[1]/@href, "pdf")'
-        self.cell_path = '//table[@id="decisions"]//tr[' + limiter + "]/td[%d]"
+        self.url = (
+            "https://www.illinoiscourts.gov/top-level-opinions?type=supreme"
+        )
 
-    def _get_download_urls(self):
-        path = "%s//a[1]/@href" % (self.cell_path % 5)
-        return [href for href in self.html.xpath(path)]
+    def _process_html(self):
+        date_amend_rule_23 = datetime.date(2021, 1, 1)
+        for row in self.html.xpath("//table[@id='ctl04_gvDecisions']//tr"):
+            cells = row.xpath(".//td")
+            if len(cells) == 7:
+                try:
+                    name = get_row_column_text(row, 1)
+                    citation = get_row_column_text(row, 2)
+                    date = get_row_column_text(row, 3)
+                    url = get_row_column_links(row, 1)
+                    # After 2021-01-01, all documents are Precedential
+                    if (
+                        datetime.datetime.strptime(
+                            date.strip(), "%m/%d/%Y"
+                        ).date()
```

Comment:
I think this will work without the cast to date (by creating a date in the first place):
```suggestion
                        datetime.date.strptime(
                            date.strip(), "%m/%d/%Y"
                        )
```

We also have some utils for this in, I think, date_utils.py, that might make it easier/tidier.

Reply:
Done

Reply:
Al requested changes are done.
Here's a sample os the output from the manual execution of the scrapers:
For `python sample_caller.py -vvv -c juriscraper.opinions.united_states.state.ill` :
```
Adding new item:
    case_dates: 2021-05-20
    case_names: "People v. Bochenek"
    download_urls: "https://www.illinoiscourts.gov/resources/f515c232-1a6b-4f2b-a033-3e4c81bb9cd0/file"
    precedential_statuses: "Published"
    blocked_statuses: False
    date_filed_is_approximate: False
    docket_numbers: "125889"
    neutral_citations: "2021 IL 125889"
    case_name_shorts: "Bochenek"

Adding new item:
    case_dates: 2021-05-20
    case_names: "Palos Community Hospital v. Humana Insurance Co., Inc."
    download_urls: "https://www.illinoiscourts.gov/resources/75ea3309-6505-4026-9f41-c3ab19592008/file"
    precedential_statuses: "Published"
    blocked_statuses: False
    date_filed_is_approximate: False
    docket_numbers: "126008"
    neutral_citations: "2021 IL 126008"
    case_name_shorts: ""
```
For `python sample_caller.py -vvv -c juriscraper.opinions.united_states.state.illappct` :
```
Adding new item:
    case_dates: 2021-10-20
    case_names: "Ely v. Pivar"
    download_urls: "https://www.illinoiscourts.gov/resources/d9271fba-b267-4bff-83b4-a8253ebbba33/file"
    precedential_statuses: "Published"
    blocked_statuses: False
    date_filed_is_approximate: False
    docket_numbers: "1-20-0852"
    neutral_citations: "2021 IL App (1st) 200852-U"
    case_name_shorts: "Ely"

Adding new item:
    case_dates: 2021-10-20
    case_names: "Dustman v. Advocate Aurora Health, Inc."
    download_urls: "https://www.illinoiscourts.gov/resources/2f934c8e-7dfa-4fac-b41b-0a1fde5af694/file"
    precedential_statuses: "Published"
    blocked_statuses: False
    date_filed_is_approximate: False
    docket_numbers: "4-21-0157"
    neutral_citations: "2021 IL (4th) 210157"
    case_name_shorts: "Dustman"
```